aep/general/0004/aep.md

```diff
@@ -0,0 +1,101 @@
+# Resource types
+
+Most APIs expose _resources_ (their primary nouns) which users are able to
+create, retrieve, and manipulate. APIs are allowed to name their resource types
+as they see fit, and are only required to ensure uniqueness within that API.
+This means that it is possible (and often desirable) for different APIs to use
+the same type name. For example, a Memcache and Redis API would both want to
+use `Instance` as a type name.
+
+When mapping the relationships between APIs and their resources, however, it
+becomes important to have a single, globally-unique type name. Additionally,
+tools such as Kubernetes or GraphQL interact with APIs from multiple providers.
+
+## Guidance
+
+APIs **must** define a resource type for each resource in the API, according to
+the following pattern: `{API Name}/{Type}`. The type name:
+
+- **must** Start with an uppercase letter.
+- **must** Only contain alphanumeric characters.
+- **must** Be of the singular form of the noun.
+- **must** Use PascalCase (UpperCamelCase).
+- For Kubernetes, the type name **must** match the [object][] name.
+- For Protobuf, the type name **must** match the name of the protobuf message.
+
+### Examples
+
+Examples of resource types include:
+
+- `pubsub.googleapis.com/Topic`
+- `pubsub.googleapis.com/Subscription`
+- `spanner.googleapis.com/Database`
+- `spanner.googleapis.com/Instance`
+- `networking.istio.io/Instance`
+
+### Annotating resource types
+
+APIs **must** annotate the resource types for each resource in the API using
+the [`google.api.resource`][resource] annotation:
```

Comment:
I'd rather focus on proto for now and see how things like in OAS later at the end - primarily because I have the aepc repository which just uses what grpc-proxy translates things as.

How about an issue on it for later?

Reply:
Hmm...we use proto at Roblox so it doesn't affect me in that sense, but I wonder how our OAS-aligned members will feel about OAS being effectively a secondary consideration. :/

Reply:
I'll leave it open, we can raise concerns there. Personally I'd like to see if we can find a better way to have dual support for OAS / proto than spelling out every annotation on each (although maybe that's the way it has to be).

Reply:
added OAS annotations.